app/javascript/app/ssn-field.js

```diff
@@ -38,13 +21,33 @@ function formatSSNField() {
       input.insertAdjacentHTML('afterend', el);
 
       const toggle = document.getElementById(`ssn-toggle-${i}`);
-      let ssnCleave;
-
-      ssnCleave = sync(input, toggle, ssnCleave);
 
-      toggle.addEventListener('change', function () {
-        ssnCleave = sync(input, toggle, ssnCleave);
-      });
+      let cleave;
+
+      function sync() {
+        const { value } = input;
+        input.type = toggle.checked ? 'text' : 'password';
+        cleave?.destroy();
+        if (toggle.checked) {
+          cleave = new Cleave(input, {
+            numericOnly: true,
+            blocks: [3, 2, 4],
+            delimiter: '-',
+          });
+        } else {
+          const nextValue = value.replace(/-/g, '');
+          if (nextValue !== value) {
+            input.value = nextValue;
+          }
+        }
+        const didFormat = input.value !== value;
+        if (didFormat) {
+          input.checkValidity();
```

Comment:
Not directly related, but do we need to set aria-invalid? Like, does this function still get called on these SSN fields? https://github.com/18F/identity-idp/pull/4309/files#diff-a6bced026c54f99f198462504947fc41711354db5f9045c8d26a63fdb30f2cb2

Reply:
That code runs in response to the `invalid` event, so it would at least run in response to a `checkValidity` which results in a validation error. But it doesn't seem clear to me that it would run if it needed to set `aria-invalid="false"`.

We could either repeat it here, or try to find a way to emulate `input`† event to trigger it, or find another event to be binding to in `validateInput`.

----

†

```js
input.dispatchEvent(new InputEvent('input'));
```

Reply:
Added ticket [here](https://cm-jira.usa.gov/browse/LG-3699) to follow up on this